Add user management functions

jjm opened the pull request:
Add actions for user management (Fixes: #15):
  - [x] `list_users`.
  - [x] `create_user`.
  - [ ] `delete_user`.

*WIP: creating PR to run CI checks, got one last action to write.*

blag commented on actions/delete_user.py:
```diff
@@ -0,0 +1,36 @@
+class DeleteUser(PagerDutyAction):
+    def run(self, email):
+        """
+        Delete a user from PagerDuty (`or raise an error if not possible)
```
Nitpick: errant backtick.

jjm replied:
Cheers, removed.

jjm commented:
@nmaludy Sure I'll add them. I'll see how the base class copes with a simple unittest, I've got the feeling that I'll need to refactor part of it to make an `assertIsInstance` test pass.

jjm commented:
I'll correct the failing tests and coverage for the actions later in the week when I've some free time.

jjm commented:
@nmaludy I've addressed your comments, by adding unit tests for my new actions and some basic ones for the existing actions. Plus added a section to the `README`.

blag commented on tests/test_action_ack_incident.py:
```diff
@@ -0,0 +1,11 @@
+from st2tests.base import BaseActionTestCase
+
+
+class PagerDuytyAckIncidentActionTestCase(BaseActionTestCase):
```
Nitpick: `Duyty` - @jjm Not sure if you want to correct this (and the rest of the testcase names). It's only in the name of the testcases, so it doesn't affect anything user-facing.

jjm replied:
Boo! That’s annoying.  I’ll correct that later on tonight.

blag commented:
I'm not sure what these are testing, or how they magically figure out what type of object to return:
* `tests/test_action_ack_incident.py`
* `tests/test_action_launch_incident.py`
* `tests/test_action_open_incidents.py`
* `tests/test_action_resolve_incident.py`

blag commented on actions/delete_user.py:
```diff
@@ -0,0 +1,36 @@
+                    users[0].remove()
+                except BadRequest as e:
+                    success = False
+                    results['error'] = str(e)
```
Why not just return `e`? That way you don't lose any information from it.

jjm commented:
@blag These 4 actions 

- `tests/test_action_ack_incident.py`
- `tests/test_action_launch_incident.py`
- `tests/test_action_open_incidents.py`
- `tests/test_action_resolve_incident.py`

already existed in the PagerDuty pack before this PR, so as I was adding tests for my new actions I added just a simple test to ensure the class would successfully initialise.

I'm thinking of another PR to use a new PagerDuty (and not community library) and if I do that I'll expand these tests.

blag commented:
LGTM.
